File: src/main/kotlin/org/kravbank/java/model/Project.java
package org.kravbank.java.model;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import java.util.List;

//@AllArgsConstructor
//@NoArgsConstructor
//@Data
@Entity
//@Builder
public class Project extends PanacheEntity {

//    @Column(length = 200)
    public String title;

    public String description;

    public Long version;

    public String publishedDate;

    public String projectId; //uuid

    public String deletedDate;

    @OneToMany
    public List <Publication> publications;

    @OneToMany
    public List <Need> needs;

    @OneToMany
    public List <Requirement> requirements; // -> private String requirementVarian; --> private string config -->

    @OneToMany
    public List <Product> products;

    @OneToMany

    public List <Code> codeList; //CONFIG


    //private String dependency; //vent med bruk
    //private String tags; //vent med bruk

}

File: src/main/kotlin/org/kravbank/domain/Project.kt
package org.kravbank.domain

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import io.quarkus.hibernate.orm.panache.PanacheEntity
import java.util.*
import javax.persistence.CascadeType
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.OneToMany


@Entity
@JsonIgnoreProperties(ignoreUnknown = true)
data class Project(

    @Column(
        unique = true,
        name = "ref"
    )
    var ref: String = UUID.randomUUID().toString(),

    var title: String = "",

    var description: String = "",

    @OneToMany(
        mappedBy = ("project"),
        cascade = [CascadeType.ALL],
        orphanRemoval = true,
    )
    var products: MutableList<Product>? = null,

    @OneToMany(
        mappedBy = ("project"),
        cascade = [CascadeType.ALL],
        orphanRemoval = true,
    )
    var publications: MutableList<Publication>? = null,

    @OneToMany(
        mappedBy = ("project"),
        cascade = [CascadeType.ALL],
        orphanRemoval = true,
    )
    var requirements: MutableList<Requirement>? = null,

    @OneToMany(
        mappedBy = ("project"),
        cascade = [CascadeType.ALL],
        orphanRemoval = true,
    )
    var needs: MutableList<Need>? = null,

    @OneToMany(
        mappedBy = ("project"),
        cascade = [CascadeType.ALL],
        orphanRemoval = true
    )
    var codelist: MutableList<Codelist>? = null

) : PanacheEntity() {

    override fun toString(): String {
        return "Project(title='$title', description='$description', ref='$ref', products=$products, publications=$publications, requirements=$requirements, needs=$needs, codelist=$codelist)"
    }
}